### PR TITLE
Project generation enhancements

### LINKED ISSUE #871: Yarn template still uses npm in README.md and package.json
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behaviour?**
Generated `README.md` and `package.json` scripts use `npm` instead of `yarn` even when the `--yarn` option is specified.

**If the current behaviour is a bug, please provide the steps to reproduce.**

```bash
preact create yarnapp --yarn
grep npm ./yarnapp/README.md ./yarnapp/package.json
```

**What is the expected behaviour?**

Yarn is used throughout project instructions and in `package.json`

**If this is a feature request, what is motivation or use case for changing the behaviour?**

N/A

**Please mention other relevant information.**

N/A

Please paste the results of `preact info` here.

```
Environment Info:

  System:
    OS: macOS 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i7-6567U CPU @ 3.30GHz
  Binaries:
    Node: 12.6.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 67.0.4
    Safari: 12.1.2
  npmPackages:
    preact: ^8.3.1 => 8.5.2 
    preact-cli: ^3.0.0-next.19 => 3.0.0-rc.4 
    preact-compat: ^3.18.4 => 3.19.0 
    preact-render-to-string: ^4.1.0 => 4.1.0 
    preact-router: ^2.6.1 => 2.6.1 
  npmGlobalPackages:
    preact-cli: 3.0.0-rc.4
```

### PR DESCRIPTION
**Pull request by DenysVuika**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
no, manual tests only

**Summary**
Fixes #871 

* Adds missing default values for `preact create --help`
* Moves "dirExits" (create command) to the utils
* Adds support for `--license` option for `create` (defaults to `MIT`)
* Introduces support for additional template variables:
  * `{{ pkg-install }}` => `npm install` or `yarn`
  * `{{ pkg-run }}` => `npm run` or `yarn`
  * `{{ pkg-add }}` => `npm install` or `yarn add`
  * `{{ now-year }}` => current year
  * `{{ license }}` => license or `MIT`

Example for new options:

```sh
preact create default yarnapp --yarn --install=false --license=Apache
```

**Does this PR introduce a breaking change?**
no

**Other information**
See also: https://github.com/preactjs-templates/default/pull/29

### CODE REVIEW

File: packages/cli/lib/util.js [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+function dirExits(workingDir, destDir) {
+	if (workingDir && destDir) {
+		const target = resolve(workingDir, destDir);
+		return isDir(target);
+	}
+	return false;
+}
+
+exports.isDir = isDir;
+exports.dirExists = dirExits;
```
Comment (ForsakenHarmony) [inline packages/cli/lib/util.js]: for consistency with the rest we probably want the export inline

Comment (DenysVuika) [reply packages/cli/lib/util.js]: dirExists calls isDir, should I reference it via exports or leave as is?

Comment (DenysVuika) [reply packages/cli/lib/util.js]: thanks, updated

**Comment by ForsakenHarmony** [comment]
Thanks for your contributions 🙏

**Comment by DenysVuika** [comment]
No problem 😊

**Pull request merged**